src/main/java/com/opentable/versionedconfig/GitOperations.java

```diff
@@ -85,33 +94,49 @@ private Git openRepo(final GitProperties serviceConfig, Path checkoutDir)
         } else {
             LOG.info("checkout directory {} does not yet exist", checkoutDir);
         }
-        final String cloneSource = cloningUriToGitArgument(serviceConfig.getRemoteRepository());
-        final String cloneBranch = serviceConfig.getBranch();
-        LOG.info("cloning {} (branch {}) to {}", cloneSource, cloneBranch, checkoutDir);
-
-        try {
-            CloneCommand clone = Git.cloneRepository()
-                    .setBare(false)
-                    .setBranch(cloneBranch)
-                    .setDirectory(checkoutDir.toFile())
-                    .setURI(cloneSource);
-            credentials.ifPresent(clone::setCredentialsProvider);
-            return clone.call();
-        } catch (GitAPIException ioe) {
-            throw new VersioningServiceException("Could not clone repo", ioe);
+        final List<URI> remotes = config.getRemoteRepositories();
+        final Git result = upstreamRetry(idx -> {
+            final String cloneBranch = serviceConfig.getBranch();
+            LOG.info("cloning {} (branch {}) to {}", idx, cloneBranch, checkoutDir);
+
+            try {
+                CloneCommand clone = Git.cloneRepository()
+                        .setBare(false)
+                        .setBranch(cloneBranch)
+                        .setDirectory(checkoutDir.toFile())
+                        .setURI(remotes.get(idx).toString());
+                credentials.ifPresent(clone::setCredentialsProvider);
+                return clone.call();
+            } catch (GitAPIException ioe) {
+                throw new VersioningServiceException("Could not clone repo", ioe);
+            }
+        });
+        for (int i = 0; i < remotes.size(); i++) {
+            try {
+                result.remoteAdd()
+                    .setName("remote" + i)
+                    .setUri(new URIish(remotes.get(i).toString()))
```

Comment:
I was thinking `URIish` was a @stevenschlansker-ism.

Reply:
nope, it's a Git-ism :)